lib/transaction/input/input.js

```diff
@@ -105,14 +114,19 @@ Input.prototype.toBufferWriter = function(writer) {
 };
 
 Input.prototype.setScript = function(script) {
+  this._script = null;
   if (script instanceof Script) {
     this._script = script;
     this._scriptBuffer = script.toBuffer();
+  } else if (JSUtil.isHexa(script)) {
+    // hex string script
+    this._scriptBuffer = new Buffer(script, 'hex');
```

Comment:
No, it's only set where it's already there. In this case, it will be lazy loaded if requested